### ISSUE TITLE
File index_m.html not found

### ISSUE THREAD

**Issue by lkathke** [iobroker-community-adapters/ioBroker.find-my-iphone#19]
Hello guys!
I am currently getting "File index_m.html not found" in the adapter configuration,
and also this error log:

host.328d02ae42a0 | 2020-04-16 12:26:18.717 | info | Restart adapter system.adapter.find-my-iphone-community.0 because enabled
-- | -- | -- | --
host.328d02ae42a0 | 2020-04-16 12:26:18.717 | error | instance system.adapter.find-my-iphone-community.0 terminated with code 1 (JS_CONTROLLER_STOPPED)
host.328d02ae42a0 | 2020-04-16 12:26:18.717 | error | Caught by controller[0]: at bootstrap_node.js:625:3
host.328d02ae42a0 | 2020-04-16 12:26:18.717 | error | Caught by controller[0]: at startup (bootstrap_node.js:204:16)
host.328d02ae42a0 | 2020-04-16 12:26:18.717 | error | Caught by controller[0]: at Function.Module.runMain (module.js:694:10)
host.328d02ae42a0 | 2020-04-16 12:26:18.717 | error | Caught by controller[0]: at Function.Module._load (module.js:498:3)
host.328d02ae42a0 | 2020-04-16 12:26:18.717 | error | Caught by controller[0]: at tryModuleLoad (module.js:506:12)
host.328d02ae42a0 | 2020-04-16 12:26:18.716 | error | Caught by controller[0]: at Module.load (module.js:566:32)
host.328d02ae42a0 | 2020-04-16 12:26:18.716 | error | Caught by controller[0]: at Object.Module._extensions..js (module.js:664:10)
host.328d02ae42a0 | 2020-04-16 12:26:18.716 | error | Caught by controller[0]: at Module._compile (module.js:653:30)
host.328d02ae42a0 | 2020-04-16 12:26:18.716 | error | Caught by controller[0]: at Object.<anonymous> (/opt/iobroker/node_modules/iobroker.find-my-iphone-community/find-my-iphone.js:17:20)
host.328d02ae42a0 | 2020-04-16 12:26:18.716 | error | Caught by controller[0]: at Object.exports.Adapter (/opt/iobroker/node_modules/soef/soef.js:1383:23)
host.328d02ae42a0 | 2020-04-16 12:26:18.716 | error | Caught by controller[0]: TypeError: fns.adapter is not a function
host.328d02ae42a0 | 2020-04-16 12:26:18.716 | error | Caught by controller[0]: ^
host.328d02ae42a0 | 2020-04-16 12:26:18.716 | error | Caught by controller[0]: fns.adapter = fns.adapter(options);
host.328d02ae42a0 | 2020-04-16 12:26:18.716 | error | Caught by controller[0]: /opt/iobroker/node_modules/soef/soef.js:1383

My ioBroker.admin version is 3.7.8 and I am using the current master branch (eee7f46).

Thank you in advance,
lkathke

**Comment by Apollon77** [comment]
Please run "iobroker upload find-my-iphone-community" ... is it then there?

**Comment by Tidle999** [comment]
same problem for me, also after running the command

**Comment by lkathke** [comment]
> Please run "iobroker upload find-my-iphone-community" ... is it then there?

Okay, i used the iobroker upload find-my-iphone-community and now I can see the configuration page. But there is some javascript error:

![screenshot](https://user-images.githubusercontent.com/2429447/79452205-0b7b6d80-7fe8-11ea-82ca-30379f6f5f2d.png)

Maybe I have to update my iobroker setup to a newer version.

**Comment by Tidle999** [comment]
i have the newest version and still these errors. or even worse still the index_m.html error.

**Comment by DutchmanNL** [comment]
> i have the newest version and still these errors. or even worse still the index_m.html error.

how did you install ?

**Comment by Tidle999** [comment]
adapter --> install from own url --> copied the github url --> then seen the new entry and installed:
![image](https://user-images.githubusercontent.com/36228711/79459007-ed673a80-7ff2-11ea-9685-3bb3025aa892.png)

**Comment by DutchmanNL** [comment]
> adapter --> install from own url --> copied the github url --> then seen the new entry and installed:
> ![image](https://user-images.githubusercontent.com/36228711/79459007-ed673a80-7ff2-11ea-9685-3bb3025aa892.png)

so stop doing that and use the proper way and install it by the admin and NOT by git !!!!

tip : remove all of them now, run an update (the pillow at the top) and install the version shown.

Github / URL installations ==> ALPHA never use in PROD !

**Comment by Tidle999** [comment]
the version shown (you mean 0.2.15?) is not working and seems not to be maintained any more. thats why i installed this version here. so i dont know what you mean with install by admin?

**Comment by DutchmanNL** [comment]
> the version shown (you mean 0.2.15?) is not working and seems not to be maintained any more. thats why i installed this version here. so i dont know what you mean with install by admin?

I would advise to start here : https://www.iobroker.net/#en/documentation/admin/adapter.md

Do not install this adapter from git ! development in progress and test-failing so NOT working at all

**Comment by lkathke** [comment]
There is only ioBroker.find-my-iphone available, not ioBroker.find-my-iphone-community.

Is ioBroker.find-my-iphone the stable version of ioBroker.find-my-iphone-community or the not maintained version?

**Comment by Tidle999** [comment]
yes exaclty. find-my-iphone-community is not available via npm installation, so we can just install via github

**Comment by lkathke** [comment]
It's working for me, when I install
Find my iPhone version 0.2.15
and running npm install soef@0.2.1 in /opt/iobroker

**Comment by Tidle999** [comment]
i am also using this docker and version 0.2.15 seems to work, but does not refresh the devices, i think this realease is outdated

**Comment by lkathke** [comment]
What do you mean with "refresh the devices"? Can't you see your device or is the location wrong/old?

**Comment by Tidle999** [comment]
i can see the devices, but location and battery level is not refreshing, even if i am triggering and alert is not working (or i dont know how to use), but i want to use find my iphone to let it ring. is this possible for you?

**Comment by lkathke** [comment]
When you set "find-my-iphone.0/YOUR_DEVICE_NAME/refresh" to true it will refresh your devices location. Alert also works with my phone, but I can't see the message, you can define.

**Comment by DutchmanNL** [comment]
> yes exaclty. find-my-iphone-community is not available via npm installation, so we can just install via github

Because it does not exist, it is. It working and unreleased code

What sense has it to discuss a version “in progress” not released ?!

**Comment by Tidle999** [comment]
i set it to true already, but also do not work :(

**Comment by Tidle999** [comment]
> > yes exaclty. find-my-iphone-community is not available via npm installation, so we can just install via github
> 
> Because it does not exist, it is not working and unreleased code
> 
> What sense has it to discuss a version “in progress” not released ?!

ok, we just want to give feedback ;)

**Comment by DutchmanNL** [comment]
> ok, we just want to give feedback ;)

Which is great please don’t get me wrong ;)

But, I am also a developer, I cannot troubleshoot or assist in error of code which are not released but in progress basically that takes me away from coding 😂

We have to redesign this one 

@ldittmar81 what about your progress do you think you can finalize it or ?

**Comment by kevinlieser** [comment]
Okay, so whats correct here? I have also installed the GitHub package – (not everything on GitHub is "in progress" (especially not master branches). I also thought the find-my-iphone package is no maintained anymore and this one is the replacing package.

Since I got the same error as above I came here.

So when I use the "correct" "iobroker.find-my-iphone" npm package I also got an error like "Error: Cannot find module '/opt/iobroker/node_modules/soef/../iobroker.admin/lib/utils'" and the adapter will not start.

So basically both (GitHub and npm package) do not work.

**Comment by LuxBibi** [comment]
> Okay, so whats correct here? I have also installed the GitHub package – (not everything on GitHub is "in progress" (especially not master branches). I also thought the find-my-iphone package is no maintained anymore and this one is the replacing package.
> 
> Since I got the same error as above I came here.
> 
> So when I use the "correct" "iobroker.find-my-iphone" npm package I also got an error like "Error: Cannot find module '/opt/iobroker/node_modules/soef/../iobroker.admin/lib/utils'" and the adapter will not start.
> 
> So basically both (GitHub and npm package) do not work.

Hi kevinlieser ...
Were aou able to get this working ?

If not, then I do not really understand why this "Find-My-iPhone" adapter remains in the available adapter list of iobroker. This should be removed, as it does no more work. 

Thx for your reply,
Luc

**Comment by AFPass** [comment]
hello, there are still developments here on the adapter

**Comment by clmpng** [comment]
I made the adapter work. Basically there are two things that have to be changed: 

1. issue: index_m.html not found.
    - can be solved by using "iobroker upload find-my-iphone-community"

2. issue: Cannot read property 'getState' of undefined.
-  adapter.objects/adapter.states methdos needs to be replaced by direct adapter methods. e.g.: instead of _`adapter.states.getState`_ use `_adapter.getState_`

Now, it is working.

Christian

**Comment by AFPass** [comment]
@clmpng Thank you. But what exactly did you change in soef.js? I am not a JS developer and unfortunately I am not familiar with it.

**Comment by clmpng** [comment]
> @clmpng Thank you. But what exactly did you change in soef.js? I am not a JS developer and unfortunately I am not familiar with it.

Basically, I just replaced each `adapter.states` and `adapter.objects` by `adapter`. So, for example `adapter.states.getState` becomes `adapter.getState`

**Comment by AFPass** [comment]
@clmpng Thanks. :-) It works.

**Comment by theAnDrO40** [comment]
Didn't work for me. Upload makes no changes and make changes in soef.js without any results

**Comment by karlheinz0502** [comment]
Did not work for me, too - changes in soef.js  - no results! Stable Version 0.2.15

**Comment by vonCartman** [comment]
@clmpng can you describe with more precision how did you solve the problem with "index_m.html" ??

**Comment by dieckfr** [comment]
If there a working fix for the issue???

**Comment by Goeste** [comment]
Hi,

I now run into the same problem, that index_m.html is not found. uploading did not help (with and without -community), changes in the soef.js didn't do anyhing either...

Any news on this?

Beswt,
goeste

`<html>
<body>


find-my-iphone-community.0 | 2022-03-22 13:36:48.051 | error | Cannot read properties of undefined (reading 'common')
-- | -- | -- | --
find-my-iphone-community.0 | 2022-03-22 13:36:48.050 | error | TypeError: Cannot read properties of undefined (reading 'common') at new Adapter (/opt/iobroker/node_modules/@iobroker/js-controller-adapter/build/lib/adapter/adapter.js:8474:47) at Object.Adapter [as adapter] (/opt/iobroker/node_modules/@iobroker/js-controller-adapter/build/lib/adapter/adapter.js:49:16) at Object.exports.Adapter (/opt/iobroker/node_modules/iobroker.find-my-iphone/node_modules/soef/soef.js:1383:23) at Object.<anonymous> (/opt/iobroker/node_modules/iobroker.find-my-iphone/find-my-iphone.js:17:20) at Module._compile (node:internal/modules/cjs/loader:1101:14) at Object.Module._extensions..js (node:internal/modules/cjs/loader:1153:10) at Module.load (node:internal/modules/cjs/loader:981:32) at Function.Module._load (node:internal/modules/cjs/loader:822:12) at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12) at node:internal/main/run_main_module:17:47
find-my-iphone-community.0 | 2022-03-22 13:36:48.047 | error | uncaught exception: Cannot read properties of undefined (reading 'common')
find-my-iphone-community.0 | 2022-03-22 13:36:16.393 | error | Cannot read properties of undefined (reading 'common')


</body>
</html>`

**Comment by Apollon77** [comment]
A new adapter will replace this one. Is already in beta repo. Try this please https://github.com/PfisterDaniel/ioBroker.apple-find-me
